contracts/pre-propose/cwd-pre-propose-multiple/src/tests.rs

```diff
@@ -0,0 +1,1421 @@
+use cosmwasm_std::{coins, from_slice, to_binary, Addr, Coin, Decimal, Empty, Uint128};
+use cpm::query::ProposalResponse;
+use cw2::ContractVersion;
+use cw20::Cw20Coin;
+use cw_denom::UncheckedDenom;
+use cw_multi_test::{App, BankSudo, Contract, ContractWrapper, Executor};
+use cw_utils::Duration;
+use cwd_core::state::ProposalModule;
+use cwd_interface::{Admin, ModuleInstantiateInfo};
+use cwd_pre_propose_base::{error::PreProposeError, msg::DepositInfoResponse, state::Config};
+use cwd_proposal_multiple as cpm;
+use cwd_testing::helpers::instantiate_with_cw4_groups_governance;
+use cwd_voting::{
+    deposit::{CheckedDepositInfo, DepositRefundPolicy, DepositToken, UncheckedDepositInfo},
+    multiple_choice::{
+        CheckedMultipleChoiceOption, MultipleChoiceOption, MultipleChoiceOptionType,
+        MultipleChoiceOptions, MultipleChoiceVote, VotingStrategy,
+    },
+    pre_propose::{PreProposeInfo, ProposalCreationPolicy},
+    status::Status,
+    threshold::PercentageThreshold,
+};
+
+use crate::contract::*;
+
+fn cw_dao_proposal_multiple_contract() -> Box<dyn Contract<Empty>> {
+    let contract = ContractWrapper::new(
+        cpm::contract::execute,
+        cpm::contract::instantiate,
+        cpm::contract::query,
+    )
+    .with_reply(cpm::contract::reply);
+    Box::new(contract)
+}
+
+fn cw_pre_propose_base_proposal_single() -> Box<dyn Contract<Empty>> {
+    let contract = ContractWrapper::new(execute, instantiate, query);
+    Box::new(contract)
+}
+
+fn cw20_base_contract() -> Box<dyn Contract<Empty>> {
+    let contract = ContractWrapper::new(
+        cw20_base::contract::execute,
+        cw20_base::contract::instantiate,
+        cw20_base::contract::query,
+    );
+    Box::new(contract)
+}
+
+fn get_default_proposal_module_instantiate(
```

Comment:
really nice to have all these methods to modify instead of editing a bunch of lines of test files. we should move these to `packages/testing` once we get started on red teaming